teps/0060-remote-resource-resolution.md

```diff
@@ -239,13 +242,106 @@ users can leverage those shared pipelines.
+to put their Tekton Pipelines controller in to.
+
+#### Alternative
+
+A different approach that would work for PipelineRuns today would be to require
+them to be created with a [`Pending`
+status](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#pending-pipelineruns).
+
+This proposal doesn't take that approach because:
+1. The same support does not exist for TaskRuns and so would require its own
+   independent TEP and review cycle to implement.
+2. Pending status can't be used by multiple controllers that all need to signal
+   readiness. This means a Resolution Reconciler and (for example) a custom
+   PipelineRun Scheduler could set this flag before the other controller is
+   ready to do so.
+
+### Proposal 3: Add `status.taskYaml` to `TaskRuns` and `status.pipelineYaml` to `PipelineRuns`
+
+When the proposed experimental resolution controller sees an unresolved
+`TaskRun` or `PipelineRun` it will read the resource and then resolve whatever
+ref, bundle or inline document is included. The resolved `spec` will then be
+embedded in the `status.taskYaml.spec` field for `TaskRuns` or the
+`status.pipelineYaml.spec` field for `PipelineRuns`. Once those fields are populated
+Tekton Pipeline's reconcilers will be able to pick them up and begin working
+with the specs.
+
+We will take this approach because it's entirely backwards-compatible with our
+existing resolution and reconcilers: today, the reconcilers already copy
+resolved resources into the `status` block of `TaskRuns` and `PipelineRuns` so
+that changes in the underlying resources don't affect runs that are already
+in-flight. By observing the same protocol it means that the logic changes in
+our Tekton Pipelines' reconcilers is minimized and the same "audit trail"
+property is maintained.
+
+Unfortunatley there's a wrinkle: our existing reconcilers copy labels and
+annotations from the resolved `Task` or `Pipeline` to the `TaskRun` or
+`PipelineRun`. By splitting the resolution logic in to its own external
+machinery we are introducing a barrier between resolution and metadata
+duplication: the "actuation reconcilers" never see the fully-resolved resource,
+only the `spec` passed via the `status`, so they're unable to perform the
+metadata copy. The end result is that the "resolver reconcilers" are also
+responsible for performing this metadata copy.
+
+A lot of the conversation with this TEP so far has revolved around the fact
+that resolution might be something we want to expose not just for Pipelines but
+also other Tekton projects. Having the metadata copy performed by an external
+controller doesn't make a ton of sense in that case: metadata copying is a
+feature/requirement of Tekton Pipelines, not a general requirement for
+resolution.
+
+There are two approaches that I've considered for solutions to this:
+
+1. Change the contract such that the _entire_ resolved YAML document is passed
```

Comment:
at the risk of making this too complicated and unreliable, i wonder if we should identify some exceptions to this right off the bat

I was just looking at a Task in the dogfooding cluster and the metadata looks like this:
![image](https://user-images.githubusercontent.com/432502/130669779-5480ef95-fcff-4522-b624-79cc79cfc4c2.png)

I'm wondering if we want to identify a list of metadata fields and specific annotations we don't copy, e.g.:

- the annotation `kubectl.kubernetes.io/last-applied-configuration`
- `managedFields`

@_@ OR we start with only labels? we could pave the way for potentially embedding the entire thing by making the field able to store the entire yaml, but initially only store a certain subset that we know we need?

Reply:
Updated with a note to decide precisely how we filter the copied metadata as part of this experimental phase.